test/index.spec.ts

```diff
@@ -90,12 +92,19 @@ tape('constructor with RLP', function(tester) {
 tape('serialize', function(tester) {
   const it = tester.test
   it('should work', function(t) {
-    const account = new Account(
-      'f84602820384a056e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421a0c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470',
-    )
-    t.equal(
-      account.serialize().toString('hex'),
-      'f84602820384a056e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421a0c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470',
+    const raw = {
+      nonce: '0x01',
+      balance: '0x0042',
+      stateRoot: '0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421',
+      codeHash: '0xc5d2461236f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470',
+    }
+    const account = new Account(raw)
+    t.equals(
+      Buffer.compare(
+        account.serialize(),
+        rlp.encode([raw.nonce, raw.balance, raw.stateRoot, raw.codeHash]),
+      ),
+      0,
```

Comment:
For the reference: what is the main reasoning for this switch in constructor invocation on the test?

Reply:
I was trying to write a better test case, but also realized that we can't test our `serialize` function since it is being overwritten by `ethUtil.defineProperties`. Nonetheless I left it as a better test case, as I believe we are going to simplify the constructor in the future to just the params and not accept buffers/arrays/etc.